commerce-initializer-util/src/main/java/com/liferay/commerce/initializer/util/CommerceAccountsImporter.java

```diff
@@ -0,0 +1,259 @@
+
+		Property nameProperty = PropertyFactoryUtil.forName(
+			"twoLettersISOCode");
+
+		dynamicQuery.add(nameProperty.eq(twoLetterISOCode));
+
+		List<CommerceCountry> country =
+			_commerceCountryLocalService.dynamicQuery(dynamicQuery);
+
+		return country.get(0);
+	}
+
+	protected Organization getOrganizationsByName(
+		Map<String, Organization> organizations, String organizationName) {
+
+		return organizations.get(organizationName);
+	}
+
+	private CommerceAccount _importCommerceAccount(
+			JSONObject jsonObject, ClassLoader classLoader,
+			Map<String, Organization> organizations, String dependenciesPath,
+			ServiceContext serviceContext)
+		throws IOException, PortalException {
+
+		String name = jsonObject.getString("Name");
+		String taxId = jsonObject.getString("TaxId");
+
+		// Add Commerce Account
+
+		CommerceAccount commerceAccount =
+			_commerceAccountLocalService.addCommerceAccount(
+				name, CommerceAccountConstants.DEFAULT_PARENT_ACCOUNT_ID, null,
+				taxId, CommerceAccountConstants.ACCOUNT_TYPE_BUSINESS, true,
+				null, serviceContext);
+
+		String twoLetterISOCode = jsonObject.getString("Country");
+
+		CommerceCountry commerceCountry = getCommerceCountryByISO(
+			twoLetterISOCode);
+
+		String regionCode = jsonObject.getString("Region");
+
+		long commerceRegionId = 0;
+
+		try {
+			CommerceRegion commerceRegion =
+				_commerceRegionLocalService.getCommerceRegion(
+					commerceCountry.getCommerceCountryId(), regionCode);
+
+			commerceRegionId = commerceRegion.getCommerceRegionId();
+		}
+		catch (PortalException pe) {
+			_log.error(pe, pe);
+		}
+
+		String street1 = jsonObject.getString("Street1");
+		String city = jsonObject.getString("City");
+		String zip = jsonObject.getString("Zip");
+
+		// Add Commerce Address
+
+		_commerceAddressLocalService.addCommerceAddress(
+			commerceAccount.getModelClassName(),
+			commerceAccount.getCommerceAccountId(), commerceAccount.getName(),
+			StringPool.BLANK, street1, StringPool.BLANK, StringPool.BLANK, city,
+			zip, commerceRegionId, commerceCountry.getCommerceCountryId(),
+			StringPool.BLANK, true, false, serviceContext);
+
+		//Add Company Logo
+
+		String companyLogo = jsonObject.getString("CompanyLogo");
+
+		if (!Validator.isBlank(companyLogo)) {
+			String filePath = dependenciesPath + "images/" + companyLogo;
+
+			InputStream inputStream = classLoader.getResourceAsStream(filePath);
+
+			if (inputStream != null) {
+				File file = FileUtil.createTempFile(inputStream);
+
+				String mimeType = MimeTypesUtil.getContentType(file.getName());
+
+				FileEntry fileEntry = _dlAppLocalService.addFileEntry(
+					commerceAccount.getUserId(),
+					serviceContext.getScopeGroupId(),
+					DLFolderConstants.DEFAULT_PARENT_FOLDER_ID, file.getName(),
+					mimeType, file.getName(), null, null, file, serviceContext);
+
+				commerceAccount.setLogoId(fileEntry.getFileEntryId());
+
+				_commerceAccountLocalService.updateCommerceAccount(
```

Comment:
@alecsloan @ethanbustad We can move this logic in a protected method and avoid this update call, we just check for the logo before creating and we do pass it while creating at line 122

Reply:
addCommerceAccount does not take in logoBytes, only update does, Ideally we would upsert on line 122 passing logoBytes so that account's aren't duplicated if the site is initialized multiple times.